ext/bz2/bz2_filter.c

```diff
@@ -360,7 +360,7 @@ static php_stream_filter *php_bz2_filter_create(const char *filtername, zval *fi
 					/* How much memory to allocate (1 - 9) x 100kb */
 					zend_long blocks = zval_get_long(tmpzval);
 					if (blocks < 1 || blocks > 9) {
-						php_error_docref(NULL, E_WARNING, "Invalid parameter given for number of blocks to allocate. (" ZEND_LONG_FMT ")", blocks);
+						zend_value_error("Invalid parameter given for number of blocks to allocate. (" ZEND_LONG_FMT ")", blocks);
```

Comment:
If these are changed to exceptions, then we also need to return from the function early. Otherwise we're still going to go through the filter creation.

It should be noted that previously invalid parameters here were just ignored (with a warning) and fell back to the default value.

Reply:
I suppose if we change that a mention in UPGRADING should be added?

Reply:
These early returns are now going to leak the three allocations made above.

Reply:
TBH I'm not sure it makes a lot of sense to convert this stream filter code. I don't think stream filters usually throw?

Reply:
Yeah I just did it blindly so I'll revert those ones too